### ISSUE TITLE
Memory leak when using Kubernetes port-forwarding

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#10966]
### Description

**What happened**:

Customer is seeing their Kubernetes agents and proxies memory increase and never release when using a custom tool against Teleport Kubernetes Access which opens many port-forward connections. The customer is claiming that even after stopping the tool, Kube agent does not release connections to Kubernetes API server.

Attached are a few Go profile dumps (goroutine/cpu/heap) from 2 runs: [profile.zip](https://github.com/gravitational/teleport/files/8208555/profile.zip). Goroutine dumps show a lot of connection monitor goroutines which seems to support the theory of connections not being closed.

Number of goroutines/open FDs also seems to increase:

<img width="570" alt="Screen Shot 2022-03-08 at 10 02 54 AM" src="https://user-images.githubusercontent.com/339322/157298347-d3772416-8994-48b3-9f86-91df93d36a37.png">

<img width="568" alt="Screen Shot 2022-03-08 at 10 03 06 AM" src="https://user-images.githubusercontent.com/339322/157298382-e42be402-a69c-4590-a253-a9b942073beb.png">

<img width="573" alt="Screen Shot 2022-03-08 at 10 03 13 AM" src="https://user-images.githubusercontent.com/339322/157298396-d2b308d3-6d26-43f1-a415-70a54e24bd72.png">

Let's see if we can try to reproduce this and see if this is a bug or something specific to the customer environment.

### Server Details
- Teleport version (run `teleport version`): 8.x

**Comment by jimbishopp** [comment]
<img width="1197" alt="Screen Shot 2022-03-23 at 3 55 18 PM" src="https://user-images.githubusercontent.com/492656/159810080-01d174a3-d3b0-45ec-b501-73a9eee11d30.png">

This graph shows 50 port-forward connections being created (the first spike), then a plateau while connections were open, and finally allocations being released once the connections were closed.

I have yet to be able to reproduce this issue. I attempted with and without a Teleport Kubernetes Agent on v8 and v9. The result always looked like the graph above. I did notice a similar goroutine profile to the one attached, but it was when open port-forward connections were being serviced. Once the connections were closed the monitoring routines and memory were released.

@r0mant
